Concurrent Map read and map write

### Is this a new bug in Dekart?

- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

```
2023-06-04T19:36:28Z DBG src/server/bqjob/job.go:54 > Writing Done queryID=de8b2d89-0c19-4447-9212-8964c6a69633 reportID=7ec596a6-0b35-4951-b39e-9ab1f1026a5c
2023-06-04T19:36:28Z DBG src/server/job/job.go:148 > Removing job from store queryId=de8b2d89-0c19-4447-9212-8964c6a69633
2023-06-04T19:36:28Z DBG src/server/job/job.go:150 > lock acquired jobs=1 queryId=de8b2d89-0c19-4447-9212-8964c6a69633
2023-06-04T19:36:28Z DBG src/server/dekart/query.go:135 > Job status changed query_id=de8b2d89-0c19-4447-9212-8964c6a69633 status=5
2023-06-04T19:36:28Z DBG src/server/report/report.go:84 > Ping reportID=7ec596a6-0b35-4951-b39e-9ab1f1026a5c
2023-06-04T19:36:28Z DBG src/server/report/report.go:99 > Update subscriber rid=7ec596a6-0b35-4951-b39e-9ab1f1026a5c sequence=8
2023-06-04T19:36:28Z DBG src/server/dekart/stream.go:102 > Sending report message reportID=7ec596a6-0b35-4951-b39e-9ab1f1026a5c sequence=8
2023-06-04T19:36:28Z DBG src/server/report/report.go:84 > Ping reportID=7ec596a6-0b35-4951-b39e-9ab1f1026a5c
2023-06-04T19:36:28Z DBG src/server/report/report.go:99 > Update subscriber rid=7ec596a6-0b35-4951-b39e-9ab1f1026a5c sequence=9
2023-06-04T19:36:28Z DBG src/server/dekart/stream.go:102 > Sending report message reportID=7ec596a6-0b35-4951-b39e-9ab1f1026a5c sequence=9
2023-06-04T19:36:28Z DBG src/server/report/report.go:84 > Ping reportID=7ec596a6-0b35-4951-b39e-9ab1f1026a5c
2023-06-04T19:36:28Z DBG src/server/report/report.go:99 > Update subscriber rid=7ec596a6-0b35-4951-b39e-9ab1f1026a5c sequence=10
2023-06-04T19:36:28Z DBG src/server/dekart/stream.go:102 > Sending report message reportID=7ec596a6-0b35-4951-b39e-9ab1f1026a5c sequence=10
2023-06-04T19:36:28Z DBG src/server/storage/storage.go:152 > object is successfully uploaded DEKART_CLOUD_STORAGE_BUCKET=dekart-dev name=da39a3ee5e6b4b0d3255bfef95601890afd80709.sql
2023-06-04T19:36:28Z DBG src/server/dekart/query.go:127 > Query text updated in storage
2023-06-04T19:36:28Z DBG src/server/report/report.go:84 > Ping reportID=7ec596a6-0b35-4951-b39e-9ab1f1026a5c
2023-06-04T19:36:28Z DBG src/server/report/report.go:99 > Update subscriber rid=7ec596a6-0b35-4951-b39e-9ab1f1026a5c sequence=11
fatal error: concurrent map read and map write

goroutine 6176 [running]:
runtime.throw({0x104cb8f70?, 0x1400043d6d8?})
	/usr/local/go/src/runtime/panic.go:992 +0x50 fp=0x1400043d6f0 sp=0x1400043d6c0 pc=0x1040f00a0
runtime.mapaccess1_faststr(0x0?, 0x1?, {0x14000123f50, 0x24})
	/usr/local/go/src/runtime/map_faststr.go:22 +0x3bc fp=0x1400043d760 sp=0x1400043d6f0 pc=0x1040cf4dc
dekart/src/server/report.(*Streams).Ping.func1(0x0?, {0x14000123f50?, 0x0?})
	/Users/home/dev/dekart/src/server/report/report.go:101 +0x44 fp=0x1400043d7a0 sp=0x1400043d760 pc=0x10487ecd4
dekart/src/server/report.(*Streams).Ping.func3()
	/Users/home/dev/dekart/src/server/report/report.go:102 +0x3c fp=0x1400043d7d0 sp=0x1400043d7a0 pc=0x10487ec5c
runtime.goexit()
	/usr/local/go/src/runtime/asm_arm64.s:1263 +0x4 fp=0x1400043d7d0 sp=0x1400043d7d0 pc=0x104120a54
created by dekart/src/server/report.(*Streams).Ping
	/Users/home/dev/dekart/src/server/report/report.go:100 +0x244

goroutine 1 [chan receive, 220 minutes]:
main.main()
	/Users/home/dev/dekart/src/server/main.go:165 +0xec

goroutine 19 [select]:
go.opencensus.io/stats/view.(*worker).start(0x140001f6180)
	/Users/home/go/pkg/mod/go.opencensus.io@v0.24.0/stats/view/worker.go:292 +0x88
created by go.opencensus.io/stats/view.init.0
	/Users/home/go/pkg/mod/go.opencensus.io@v0.24.0/stats/view/worker.go:34 +0xa8

goroutine 34 [select, 220 minutes]:
github.com/desertbit/timer.timerRoutine()
	/Users/home/go/pkg/mod/github.com/desertbit/timer@v0.0.0-20180107155436-c41aec40b27f/timers.go:119 +0xac
created by github.com/desertbit/timer.init.0
	/Users/home/go/pkg/mod/github.com/desertbit/timer@v0.0.0-20180107155436-c41aec40b27f/timers.go:15 +0x28

goroutine 5 [select, 220 minutes]:
database/sql.(*DB).connectionOpener(0x140002dc5b0, {0x10524afd0, 0x140002df7c0})
	/usr/local/go/src/database/sql/sql.go:1226 +0x84
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:794 +0x184

goroutine 41 [IO wait]:
internal/poll.runtime_pollWait(0x12cd841b8, 0x72)
	/usr/local/go/src/runtime/netpoll.go:302 +0xa4
internal/poll.(*pollDesc).wait(0x140000c6880?, 0x4?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:83 +0x2c
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:88
internal/poll.(*FD).Accept(0x140000c6880)
	/usr/local/go/src/internal/poll/fd_unix.go:614 +0x1d0
net.(*netFD).accept(0x140000c6880)
	/usr/local/go/src/net/fd_unix.go:172 +0x28
net.(*TCPListener).accept(0x1400000f128)
	/usr/local/go/src/net/tcpsock_posix.go:139 +0x2c
net.(*TCPListener).Accept(0x1400000f128)
	/usr/local/go/src/net/tcpsock.go:288 +0x30
net/http.(*Server).Serve(0x140002abce0, {0x10524a570, 0x1400000f128})
	/usr/local/go/src/net/http/server.go:3039 +0x318
net/http.(*Server).ListenAndServe(0x140002abce0)
	/usr/local/go/src/net/http/server.go:2968 +0x88
main.startHttpServer(0xffffffffffffffff?)
	/Users/home/dev/dekart/src/server/main.go:132 +0x20
created by main.main
	/Users/home/dev/dekart/src/server/main.go:163 +0xdc

goroutine 43 [syscall, 220 minutes]:
os/signal.signal_recv()
	/usr/local/go/src/runtime/sigqueue.go:148 +0x30
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:23 +0x20
created by os/signal.Notify.func1.1
	/usr/local/go/src/os/signal/signal.go:151 +0x30

goroutine 6020 [IO wait]:
internal/poll.runtime_pollWait(0x12cd83fd8, 0x72)
	/usr/local/go/src/runtime/netpoll.go:302 +0xa4
internal/poll.(*pollDesc).wait(0x140000c6780?, 0x1400030e000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:83 +0x2c
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:88
internal/poll.(*FD).Read(0x140000c6780, {0x1400030e000, 0x1978, 0x1978})
	/usr/local/go/src/internal/poll/fd_unix.go:167 +0x1e4
net.(*netFD).Read(0x140000c6780, {0x1400030e000?, 0x1400030e005?, 0x6?})
	/usr/local/go/src/net/fd_posix.go:55 +0x2c
net.(*conn).Read(0x14000132020, {0x1400030e000?, 0x140000897c8?, 0x1040d1928?})
	/usr/local/go/src/net/net.go:183 +0x38
crypto/tls.(*atLeastReader).Read(0x14000632078, {0x1400030e000?, 0x0?, 0x1040c7918?})
	/usr/local/go/src/crypto/tls/conn.go:785 +0x44
bytes.(*Buffer).ReadFrom(0x14000597078, {0x10523ca58, 0x14000632078})
	/usr/local/go/src/bytes/buffer.go:204 +0x94
crypto/tls.(*Conn).readFromUntil(0x14000596e00, {0x12cc02f98?, 0x14000132020}, 0xa00000000001978?)
	/usr/local/go/src/crypto/tls/conn.go:807 +0xd8
crypto/tls.(*Conn).readRecordOrCCS(0x14000596e00, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:614 +0xdc
crypto/tls.(*Conn).readRecord(...)
	/usr/local/go/src/crypto/tls/conn.go:582
crypto/tls.(*Conn).Read(0x14000596e00, {0x1400037b000, 0x1000, 0x0?})
	/usr/local/go/src/crypto/tls/conn.go:1285 +0x180
net/http.(*persistConn).Read(0x14000fc4120, {0x1400037b000?, 0x140002f0d18?, 0x104103370?})
	/usr/local/go/src/net/http/transport.go:1929 +0x54
bufio.(*Reader).fill(0x140002b14a0)
	/usr/local/go/src/bufio/bufio.go:106 +0x100
bufio.(*Reader).Peek(0x140002b14a0, 0x1)
	/usr/local/go/src/bufio/bufio.go:144 +0x64
net/http.(*persistConn).readLoop(0x14000fc4120)
	/usr/local/go/src/net/http/transport.go:2093 +0x148
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1750 +0x11f4

goroutine 5851 [IO wait]:
internal/poll.runtime_pollWait(0x12cd83c18, 0x72)
	/usr/local/go/src/runtime/netpoll.go:302 +0xa4
internal/poll.(*pollDesc).wait(0x140003ae200?, 0x140012fe000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:83 +0x2c
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:88
internal/poll.(*FD).Read(0x140003ae200, {0x140012fe000, 0x1000, 0x1000})
	/usr/local/go/src/internal/poll/fd_unix.go:167 +0x1e4
net.(*netFD).Read(0x140003ae200, {0x140012fe000?, 0x14000351a78?, 0x104355c70?})
	/usr/local/go/src/net/fd_posix.go:55 +0x2c
net.(*conn).Read(0x14000390010, {0x140012fe000?, 0x14000351ac8?, 0x72?})
	/usr/local/go/src/net/net.go:183 +0x38
net/http.(*connReader).Read(0x14001638450, {0x140012fe000, 0x1000, 0x1000})
	/usr/local/go/src/net/http/server.go:780 +0x248
bufio.(*Reader).fill(0x140006302a0)
	/usr/local/go/src/bufio/bufio.go:106 +0x100
bufio.(*Reader).Peek(0x140006302a0, 0x4)
	/usr/local/go/src/bufio/bufio.go:144 +0x64
net/http.(*conn).serve(0x140002ac0a0, {0x10524b078, 0x14000338600})
	/usr/local/go/src/net/http/server.go:1992 +0x660
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:3071 +0x450

goroutine 6051 [select]:
google.golang.org/grpc/internal/transport.(*serverHandlerTransport).runStream(0x14000728120)
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/internal/transport/handler_server.go:425 +0x74
google.golang.org/grpc/internal/transport.(*serverHandlerTransport).HandleStreams(0x14000728120, 0x1400035c180, 0x104cb0916?)
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/internal/transport/handler_server.go:415 +0x624
google.golang.org/grpc.(*Server).serveStreams(0x1400033e000, {0x10524f690?, 0x14000728120})
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:949 +0x138
google.golang.org/grpc.(*Server).ServeHTTP(0x1400033e000, {0x105249ee0, 0x1400035c120}, 0x10489958c?)
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:1018 +0xbc
github.com/improbable-eng/grpc-web/go/grpcweb.(*WrappedGrpcServer).HandleGrpcWebRequest(0x1400038a600, {0x10524a780?, 0x140001c4000}, 0x140001f8200)
	/Users/home/go/pkg/mod/github.com/improbable-eng/grpc-web@v0.13.0/go/grpcweb/wrapper.go:146 +0x2ec
net/http.HandlerFunc.ServeHTTP(0x14000535080?, {0x10524a780?, 0x140001c4000?}, 0x140001f8200?)
	/usr/local/go/src/net/http/server.go:2084 +0x3c
github.com/rs/cors.(*Cors).Handler.func1({0x10524a780, 0x140001c4000}, 0x140001f8200)
	/Users/home/go/pkg/mod/github.com/rs/cors@v1.7.0/cors.go:219 +0x1e8
net/http.HandlerFunc.ServeHTTP(0x14000535080?, {0x10524a780?, 0x140001c4000?}, 0x104c9bf2d?)
	/usr/local/go/src/net/http/server.go:2084 +0x3c
github.com/improbable-eng/grpc-web/go/grpcweb.(*WrappedGrpcServer).ServeHTTP(0x1400038a600, {0x10524a780, 0x140001c4000}, 0x14000048168?)
	/Users/home/go/pkg/mod/github.com/improbable-eng/grpc-web@v0.13.0/go/grpcweb/wrapper.go:127 +0x190
dekart/src/server/app.Configure.func1({0x10524a780, 0x140001c4000}, 0x140001f8100)
	/Users/home/dev/dekart/src/server/app/app.go:140 +0x278
net/http.HandlerFunc.ServeHTTP(0x0?, {0x10524a780?, 0x140001c4000?}, 0x1040ca0d0?)
	/usr/local/go/src/net/http/server.go:2084 +0x3c
net/http.serverHandler.ServeHTTP({0x105248238?}, {0x10524a780, 0x140001c4000}, 0x140001f8100)
	/usr/local/go/src/net/http/server.go:2916 +0x3fc
net/http.(*conn).serve(0x140003a0140, {0x10524b078, 0x14000338600})
	/usr/local/go/src/net/http/server.go:1966 +0x56c
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:3071 +0x450

goroutine 6135 [select]:
dekart/src/server/dekart.Server.GetReportStream({0x140002dc5b0, 0x1400057e540, {0x105241af8, 0x140001f6900}, {}, {0x1052495b0, 0x140003083c0}}, 0x140002de700, {0x10524e010, 0x140003724c0})
	/Users/home/dev/dekart/src/server/dekart/stream.go:100 +0x3d0
dekart/src/proto._Dekart_GetReportStream_Handler({0x10520ed00?, 0x14000382870}, {0x10524ce90, 0x140001c40e0})
	/Users/home/dev/dekart/src/proto/dekart_grpc.pb.go:580 +0xd8
google.golang.org/grpc.(*Server).processStreamingRPC(0x1400033e000, {0x10524f690, 0x14000728120}, 0x140002d6000, 0x14000382a20, 0x105a7a140, 0x0)
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:1629 +0xf64
google.golang.org/grpc.(*Server).handleStream(0x1400033e000, {0x10524f690, 0x14000728120}, 0x140002d6000, 0x0)
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:1717 +0x814
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:965 +0x88
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:963 +0x298

goroutine 6137 [IO wait]:
internal/poll.runtime_pollWait(0x12cd83858, 0x72)
	/usr/local/go/src/runtime/netpoll.go:302 +0xa4
internal/poll.(*pollDesc).wait(0x140001f6380?, 0x140003be000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:83 +0x2c
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:88
internal/poll.(*FD).Read(0x140001f6380, {0x140003be000, 0x205, 0x205})
	/usr/local/go/src/internal/poll/fd_unix.go:167 +0x1e4
net.(*netFD).Read(0x140001f6380, {0x140003be000?, 0x0?, 0x1040d398c?})
	/usr/local/go/src/net/fd_posix.go:55 +0x2c
net.(*conn).Read(0x140000c2058, {0x140003be000?, 0x140006d0a80?, 0x2?})
	/usr/local/go/src/net/net.go:183 +0x38
crypto/tls.(*atLeastReader).Read(0x14000632060, {0x140003be000?, 0x0?, 0x1040c7918?})
	/usr/local/go/src/crypto/tls/conn.go:785 +0x44
bytes.(*Buffer).ReadFrom(0x140001e65f8, {0x10523ca58, 0x14000632060})
	/usr/local/go/src/bytes/buffer.go:204 +0x94
crypto/tls.(*Conn).readFromUntil(0x140001e6380, {0x12cc02f98?, 0x140000c2058}, 0x1040facb4?)
	/usr/local/go/src/crypto/tls/conn.go:807 +0xd8
crypto/tls.(*Conn).readRecordOrCCS(0x140001e6380, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:614 +0xdc
crypto/tls.(*Conn).readRecord(...)
	/usr/local/go/src/crypto/tls/conn.go:582
crypto/tls.(*Conn).readHandshake(0x140001e6380)
	/usr/local/go/src/crypto/tls/conn.go:1017 +0x4c
crypto/tls.(*Conn).clientHandshake(0x140001e6380, {0x10524afd0, 0x140002de780})
	/usr/local/go/src/crypto/tls/handshake_client.go:179 +0x1c0
crypto/tls.(*Conn).handshakeContext(0x140001e6380, {0x10524b040, 0x140008170e0})
	/usr/local/go/src/crypto/tls/conn.go:1460 +0x2e8
crypto/tls.(*Conn).HandshakeContext(...)
	/usr/local/go/src/crypto/tls/conn.go:1403
net/http.(*persistConn).addTLS.func2()
	/usr/local/go/src/net/http/transport.go:1537 +0x70
created by net/http.(*persistConn).addTLS
	/usr/local/go/src/net/http/transport.go:1533 +0x2d0

goroutine 6087 [IO wait]:
internal/poll.runtime_pollWait(0x12cd83b28, 0x72)
	/usr/local/go/src/runtime/netpoll.go:302 +0xa4
internal/poll.(*pollDesc).wait(0x14000375080?, 0x140001a7000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:83 +0x2c
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:88
internal/poll.(*FD).Read(0x14000375080, {0x140001a7000, 0x164e, 0x164e})
	/usr/local/go/src/internal/poll/fd_unix.go:167 +0x1e4
net.(*netFD).Read(0x14000375080, {0x140001a7000?, 0x1400008d7a0?, 0x12?})
	/usr/local/go/src/net/fd_posix.go:55 +0x2c
net.(*conn).Read(0x140000c2330, {0x140001a7000?, 0x1400008d7c8?, 0x27?})
	/usr/local/go/src/net/net.go:183 +0x38
crypto/tls.(*atLeastReader).Read(0x14000633140, {0x140001a7000?, 0x0?, 0x1040c7918?})
	/usr/local/go/src/crypto/tls/conn.go:785 +0x44
bytes.(*Buffer).ReadFrom(0x14000515778, {0x10523ca58, 0x14000633140})
	/usr/local/go/src/bytes/buffer.go:204 +0x94
crypto/tls.(*Conn).readFromUntil(0x14000515500, {0x12cc02f98?, 0x140000c2330}, 0x0?)
	/usr/local/go/src/crypto/tls/conn.go:807 +0xd8
crypto/tls.(*Conn).readRecordOrCCS(0x14000515500, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:614 +0xdc
crypto/tls.(*Conn).readRecord(...)
	/usr/local/go/src/crypto/tls/conn.go:582
crypto/tls.(*Conn).Read(0x14000515500, {0x1400063f000, 0x1000, 0x10524e5c0?})
	/usr/local/go/src/crypto/tls/conn.go:1285 +0x180
bufio.(*Reader).Read(0x1400060f620, {0x140003f32a0, 0x9, 0x14000515500?})
	/usr/local/go/src/bufio/bufio.go:236 +0x1e4
io.ReadAtLeast({0x10523bbb8, 0x1400060f620}, {0x140003f32a0, 0x9, 0x9}, 0x9)
	/usr/local/go/src/io/io.go:331 +0xa4
io.ReadFull(...)
	/usr/local/go/src/io/io.go:350
golang.org/x/net/http2.readFrameHeader({0x140003f32a0?, 0x9?, 0x14000055b00?}, {0x10523bbb8?, 0x1400060f620?})
	/Users/home/go/pkg/mod/golang.org/x/net@v0.1.0/http2/frame.go:237 +0x5c
golang.org/x/net/http2.(*Framer).ReadFrame(0x140003f3260)
	/Users/home/go/pkg/mod/golang.org/x/net@v0.1.0/http2/frame.go:498 +0x84
golang.org/x/net/http2.(*clientConnReadLoop).run(0x1400008df88)
	/Users/home/go/pkg/mod/golang.org/x/net@v0.1.0/http2/transport.go:2156 +0x100
golang.org/x/net/http2.(*ClientConn).readLoop(0x14000140900)
	/Users/home/go/pkg/mod/golang.org/x/net@v0.1.0/http2/transport.go:2051 +0x60
created by golang.org/x/net/http2.(*Transport).newClientConn
	/Users/home/go/pkg/mod/golang.org/x/net@v0.1.0/http2/transport.go:748 +0x924

goroutine 6167 [chan receive]:
net/http.(*persistConn).addTLS(0x140000f8240, {0x10524b040?, 0x140008170e0}, {0x1400004b458, 0x10}, 0x0)
	/usr/local/go/src/net/http/transport.go:1543 +0x2e0
net/http.(*Transport).dialConn(0x105a7e9c0, {0x10524b040, 0x140008170e0}, {{}, 0x0, {0x140003865d0, 0x5}, {0x1400004b458, 0x14}, 0x0})
	/usr/local/go/src/net/http/transport.go:1617 +0x790
net/http.(*Transport).dialConnFor(0x0?, 0x14000242790)
	/usr/local/go/src/net/http/transport.go:1449 +0x80
created by net/http.(*Transport).queueForDial
	/usr/local/go/src/net/http/transport.go:1418 +0x39c

goroutine 6136 [IO wait]:
internal/poll.runtime_pollWait(0x12cd83df8, 0x72)
	/usr/local/go/src/runtime/netpoll.go:302 +0xa4
internal/poll.(*pollDesc).wait(0x140001f6100?, 0x140013c6431?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:83 +0x2c
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:88
internal/poll.(*FD).Read(0x140001f6100, {0x140013c6431, 0x1, 0x1})
	/usr/local/go/src/internal/poll/fd_unix.go:167 +0x1e4
net.(*netFD).Read(0x140001f6100, {0x140013c6431?, 0x14000399f80?, 0x10523d1d8?})
	/usr/local/go/src/net/fd_posix.go:55 +0x2c
net.(*conn).Read(0x140001320a8, {0x140013c6431?, 0x0?, 0x0?})
	/usr/local/go/src/net/net.go:183 +0x38
net/http.(*connReader).backgroundRead(0x140013c6420)
	/usr/local/go/src/net/http/server.go:672 +0x48
created by net/http.(*connReader).startBackgroundRead
	/usr/local/go/src/net/http/server.go:668 +0xd0

goroutine 5850 [IO wait]:
internal/poll.runtime_pollWait(0x12cd83948, 0x72)
	/usr/local/go/src/runtime/netpoll.go:302 +0xa4
internal/poll.(*pollDesc).wait(0x14000375380?, 0x14000382791?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:83 +0x2c
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:88
internal/poll.(*FD).Read(0x14000375380, {0x14000382791, 0x1, 0x1})
	/usr/local/go/src/internal/poll/fd_unix.go:167 +0x1e4
net.(*netFD).Read(0x14000375380, {0x14000382791?, 0x0?, 0x104120a54?})
	/usr/local/go/src/net/fd_posix.go:55 +0x2c
net.(*conn).Read(0x140000100c0, {0x14000382791?, 0x14000156120?, 0x14000fc4000?})
	/usr/local/go/src/net/net.go:183 +0x38
net/http.(*connReader).backgroundRead(0x14000382780)
	/usr/local/go/src/net/http/server.go:672 +0x48
created by net/http.(*connReader).startBackgroundRead
	/usr/local/go/src/net/http/server.go:668 +0xd0

goroutine 5847 [select]:
google.golang.org/grpc/internal/transport.(*serverHandlerTransport).HandleStreams.func1()
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/internal/transport/handler_server.go:343 +0xac
created by google.golang.org/grpc/internal/transport.(*serverHandlerTransport).HandleStreams
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/internal/transport/handler_server.go:342 +0x10c

goroutine 6073 [runnable]:
dekart/src/server/report.(*Streams).Ping(0x1400057e540, {0x14000123f50, 0x24})
	/Users/home/dev/dekart/src/server/report/report.go:94 +0x1f8
dekart/src/server/dekart.Server.storeQuery({0x140002dc5b0, 0x1400057e540, {0x105241af8, 0x140001f6900}, {}, {0x1052495b0, 0x140003083c0}}, {0x14000123f50, 0x24}, {0x14000123f80, ...}, ...)
	/Users/home/dev/dekart/src/server/dekart/query.go:128 +0x1d8
created by dekart/src/server/dekart.Server.CreateQuery
	/Users/home/dev/dekart/src/server/dekart/query.go:68 +0x570

goroutine 6021 [select]:
net/http.(*persistConn).writeLoop(0x14000fc4120)
	/usr/local/go/src/net/http/transport.go:2392 +0xa4
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1751 +0x1240

goroutine 5881 [select]:
google.golang.org/grpc/internal/transport.(*serverHandlerTransport).runStream(0x14000156120)
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/internal/transport/handler_server.go:425 +0x74
google.golang.org/grpc/internal/transport.(*serverHandlerTransport).HandleStreams(0x14000156120, 0x14001638120, 0x104cb0916?)
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/internal/transport/handler_server.go:415 +0x624
google.golang.org/grpc.(*Server).serveStreams(0x1400033e000, {0x10524f690?, 0x14000156120})
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:949 +0x138
google.golang.org/grpc.(*Server).ServeHTTP(0x1400033e000, {0x105249ee0, 0x14001638090}, 0x10489958c?)
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:1018 +0xbc
github.com/improbable-eng/grpc-web/go/grpcweb.(*WrappedGrpcServer).HandleGrpcWebRequest(0x1400038a600, {0x10524a780?, 0x140003f2000}, 0x1400055a100)
	/Users/home/go/pkg/mod/github.com/improbable-eng/grpc-web@v0.13.0/go/grpcweb/wrapper.go:146 +0x2ec
net/http.HandlerFunc.ServeHTTP(0x14000535080?, {0x10524a780?, 0x140003f2000?}, 0x1400055a100?)
	/usr/local/go/src/net/http/server.go:2084 +0x3c
github.com/rs/cors.(*Cors).Handler.func1({0x10524a780, 0x140003f2000}, 0x1400055a100)
	/Users/home/go/pkg/mod/github.com/rs/cors@v1.7.0/cors.go:219 +0x1e8
net/http.HandlerFunc.ServeHTTP(0x14000535080?, {0x10524a780?, 0x140003f2000?}, 0x104c9bf2d?)
	/usr/local/go/src/net/http/server.go:2084 +0x3c
github.com/improbable-eng/grpc-web/go/grpcweb.(*WrappedGrpcServer).ServeHTTP(0x1400038a600, {0x10524a780, 0x140003f2000}, 0x14000048168?)
	/Users/home/go/pkg/mod/github.com/improbable-eng/grpc-web@v0.13.0/go/grpcweb/wrapper.go:127 +0x190
dekart/src/server/app.Configure.func1({0x10524a780, 0x140003f2000}, 0x140001f8f00)
	/Users/home/dev/dekart/src/server/app/app.go:140 +0x278
net/http.HandlerFunc.ServeHTTP(0x0?, {0x10524a780?, 0x140003f2000?}, 0x1040ca0d0?)
	/usr/local/go/src/net/http/server.go:2084 +0x3c
net/http.serverHandler.ServeHTTP({0x105248238?}, {0x10524a780, 0x140003f2000}, 0x140001f8f00)
	/usr/local/go/src/net/http/server.go:2916 +0x3fc
net/http.(*conn).serve(0x140003377c0, {0x10524b078, 0x14000338600})
	/usr/local/go/src/net/http/server.go:1966 +0x56c
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:3071 +0x450

goroutine 6045 [IO wait]:
internal/poll.runtime_pollWait(0x12cd840c8, 0x72)
	/usr/local/go/src/runtime/netpoll.go:302 +0xa4
internal/poll.(*pollDesc).wait(0x14000374580?, 0x140001a5800?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:83 +0x2c
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:88
internal/poll.(*FD).Read(0x14000374580, {0x140001a5800, 0x164d, 0x164d})
	/usr/local/go/src/internal/poll/fd_unix.go:167 +0x1e4
net.(*netFD).Read(0x14000374580, {0x140001a5800?, 0x1400062d7a0?, 0x12?})
	/usr/local/go/src/net/fd_posix.go:55 +0x2c
net.(*conn).Read(0x140000c21e0, {0x140001a5800?, 0x1400062d7c8?, 0x27?})
	/usr/local/go/src/net/net.go:183 +0x38
crypto/tls.(*atLeastReader).Read(0x14000632780, {0x140001a5800?, 0x0?, 0x1040c7918?})
	/usr/local/go/src/crypto/tls/conn.go:785 +0x44
bytes.(*Buffer).ReadFrom(0x14000597af8, {0x10523ca58, 0x14000632780})
	/usr/local/go/src/bytes/buffer.go:204 +0x94
crypto/tls.(*Conn).readFromUntil(0x14000597880, {0x12cc02f98?, 0x140000c21e0}, 0x0?)
	/usr/local/go/src/crypto/tls/conn.go:807 +0xd8
crypto/tls.(*Conn).readRecordOrCCS(0x14000597880, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:614 +0xdc
crypto/tls.(*Conn).readRecord(...)
	/usr/local/go/src/crypto/tls/conn.go:582
crypto/tls.(*Conn).Read(0x14000597880, {0x140003f5000, 0x1000, 0x10524e5c0?})
	/usr/local/go/src/crypto/tls/conn.go:1285 +0x180
bufio.(*Reader).Read(0x14000217b00, {0x140003f2820, 0x9, 0x14000597880?})
	/usr/local/go/src/bufio/bufio.go:236 +0x1e4
io.ReadAtLeast({0x10523bbb8, 0x14000217b00}, {0x140003f2820, 0x9, 0x9}, 0x9)
	/usr/local/go/src/io/io.go:331 +0xa4
io.ReadFull(...)
	/usr/local/go/src/io/io.go:350
net/http.http2readFrameHeader({0x140003f2820?, 0x9?, 0x14000054fc0?}, {0x10523bbb8?, 0x14000217b00?})
	/usr/local/go/src/net/http/h2_bundle.go:1566 +0x5c
net/http.(*http2Framer).ReadFrame(0x140003f27e0)
	/usr/local/go/src/net/http/h2_bundle.go:1830 +0x84
net/http.(*http2clientConnReadLoop).run(0x1400062df88)
	/usr/local/go/src/net/http/h2_bundle.go:8815 +0x100
net/http.(*http2ClientConn).readLoop(0x14000295c80)
	/usr/local/go/src/net/http/h2_bundle.go:8711 +0x60
created by net/http.(*http2Transport).newClientConn
	/usr/local/go/src/net/http/h2_bundle.go:7439 +0x924

goroutine 5849 [select]:
dekart/src/server/dekart.Server.GetReportStream({0x140002dc5b0, 0x1400057e540, {0x105241af8, 0x140001f6900}, {}, {0x1052495b0, 0x140003083c0}}, 0x14000384180, {0x10524e010, 0x1400188e1d0})
	/Users/home/dev/dekart/src/server/dekart/stream.go:100 +0x3d0
dekart/src/proto._Dekart_GetReportStream_Handler({0x10520ed00?, 0x14000382870}, {0x10524ce90, 0x140003f20e0})
	/Users/home/dev/dekart/src/proto/dekart_grpc.pb.go:580 +0xd8
google.golang.org/grpc.(*Server).processStreamingRPC(0x1400033e000, {0x10524f690, 0x14000156120}, 0x14000fc4000, 0x14000382a20, 0x105a7a140, 0x0)
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:1629 +0xf64
google.golang.org/grpc.(*Server).handleStream(0x1400033e000, {0x10524f690, 0x14000156120}, 0x14000fc4000, 0x0)
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:1717 +0x814
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:965 +0x88
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/server.go:963 +0x298

goroutine 5873 [IO wait]:
internal/poll.runtime_pollWait(0x12cd83a38, 0x72)
	/usr/local/go/src/runtime/netpoll.go:302 +0xa4
internal/poll.(*pollDesc).wait(0x14000398100?, 0x1400107b000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:83 +0x2c
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:88
internal/poll.(*FD).Read(0x14000398100, {0x1400107b000, 0x1000, 0x1000})
	/usr/local/go/src/internal/poll/fd_unix.go:167 +0x1e4
net.(*netFD).Read(0x14000398100, {0x1400107b000?, 0x14000587a78?, 0x104355c70?})
	/usr/local/go/src/net/fd_posix.go:55 +0x2c
net.(*conn).Read(0x140000c2020, {0x1400107b000?, 0x14000587ac8?, 0x72?})
	/usr/local/go/src/net/net.go:183 +0x38
net/http.(*connReader).Read(0x1400101e090, {0x1400107b000, 0x1000, 0x1000})
	/usr/local/go/src/net/http/server.go:780 +0x248
bufio.(*Reader).fill(0x140003f6780)
	/usr/local/go/src/bufio/bufio.go:106 +0x100
bufio.(*Reader).Peek(0x140003f6780, 0x4)
	/usr/local/go/src/bufio/bufio.go:144 +0x64
net/http.(*conn).serve(0x140001c2d20, {0x10524b078, 0x14000338600})
	/usr/local/go/src/net/http/server.go:1992 +0x660
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:3071 +0x450

goroutine 6133 [select]:
google.golang.org/grpc/internal/transport.(*serverHandlerTransport).HandleStreams.func1()
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/internal/transport/handler_server.go:343 +0xac
created by google.golang.org/grpc/internal/transport.(*serverHandlerTransport).HandleStreams
	/Users/home/go/pkg/mod/google.golang.org/grpc@v1.50.1/internal/transport/handler_server.go:342 +0x10c

goroutine 6138 [select]:
crypto/tls.(*Conn).handshakeContext.func2()
	/usr/local/go/src/crypto/tls/conn.go:1436 +0x7c
created by crypto/tls.(*Conn).handshakeContext
	/usr/local/go/src/crypto/tls/conn.go:1435 +0x184
exit status 2
make: *** [server] Error 1

```

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Relevant log output

_No response_

### Environment

```markdown
- OS:
- Dekart version:
- Browser:
```


### Which database adapter are you using with dekart?

_No response_

### Additional Context

_No response_

--- a/src/server/report/report.go
+++ b/src/server/report/report.go
@@ -97,9 +97,9 @@ func (s *Streams) Ping(reportID string) {
 		}
 		for _, ch := range streamMap {
 			log.Debug().Int64("sequence", s.sequence[rid]).Str("rid", rid).Msgf("Update subscriber")
-			go func(ch chan int64, rid string) {
-				ch <- s.sequence[rid]
-			}(ch, rid)
+			go func(ch chan int64, sequence int64) {
+				ch <- sequence
+			}(ch, s.sequence[rid])
 		}
 	}
 }

--- a/src/server/report/report_test.go
+++ b/src/server/report/report_test.go
@@ -1,0 +1,58 @@
+package report
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func waitTimeout(wg *sync.WaitGroup, timeout time.Duration) bool {
+	c := make(chan struct{})
+	go func() {
+		defer close(c)
+		wg.Wait()
+	}()
+	select {
+	case <-c:
+		return true // completed normally
+	case <-time.After(timeout):
+		return false // timed out
+	}
+}
+
+func TestReportStreamsBasicFlow(t *testing.T) {
+	streams := NewStreams()
+	reportID := "report1"
+	sequence := int64(0)
+	var wg sync.WaitGroup
+
+	numStreams := 2
+	numPings := 10
+	wg.Add(numStreams*numPings + numStreams)
+	for i := 0; i < numStreams; i++ {
+		t.Logf("Starting goroutine %d", i)
+		ch := streams.Register(reportID, fmt.Sprintf("stream%d", i), sequence)
+		go func(ch chan int64) {
+			//read from channel until it is closed
+			for s := range ch {
+				t.Logf("received from channel: %d", s)
+				wg.Done()
+			}
+
+		}(ch)
+	}
+
+	// Perform concurrent Ping operations
+	for j := 0; j < numPings; j++ {
+		go func() {
+			streams.Ping(reportID)
+		}()
+	}
+	if !waitTimeout(&wg, 3*time.Second) {
+		assert.Fail(t, "WaitGroup timed out")
+	}
+
+}

